src/Microsoft.OData.Client/Serialization/HttpWebRequestMessage.cs

```diff
@@ -214,8 +208,31 @@ public override bool SendChunked
         /// <returns>Returns the value of the header with the given name.</returns>
         public override string GetHeader(string headerName)
         {
-            Util.CheckArgumentNullAndEmpty(headerName, "headerName");
-            return HttpWebRequestMessage.GetHeaderValue(this.httpRequest, headerName);
+            if (_contentHeaderNames.Contains(headerName))
+            {
+                // If this is a "Content" header then we retrieve the value either

```

Comment:
Curious... How do we know it's a Content header?
This method will return null or empty string **so long as** `_requestMessage.Content` is not `null`.
I think we should first check if `_requestMessage.Content.Headers.GetValues(headerName)` returns anything, if it does we `string.Join` and return, otherwise allow the rest of the code to execute.

Reply:
Not 100% sure but check if `Headers` in `_requestMessage.Content` can be null. You could use the null-conditional operator...
```
IEnumerable<string> headers = _requestMessage.Content?.Headers?.GetValues(headerName);
// You could use if (headers?.FirstOrDefault() != null) since .Count() is more expensive
if (headers != null && headers.Count())
{
    return string.Join(",", headers)
}

Reply:
@gathogojr  if  requestMessage.Content is null then there is no way there can headers. It will be a content headers if it is in the message content. That is why we first check if the message content is there.. if it is there then we can check for headers. 
Also, we are checking the headers from the message content or the cache.. if the message content is null then we'll check for the header in the cache.

Reply:
@ElizabethOkerio So I understand, if _requestMessage.Content is NOT null and we don't find the header there, there's no reason to check the cache, right?